File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minCostClimbingStairs = function(cost) {
    let minCost = new Array(cost.length+1).fill(0)
    for (let i = 2; i < minCost.length; i++){
      minCost[i] = Math.min(minCost[i-1] + cost[i-1], minCost[i-2] + cost[i-2]) 
    }
    return minCost[minCost.length-1]
};

File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.cpp
#include <vector>
class Solution {
 public:
  int minCostClimbingStairs(std::vector<int> &cost) {
    std::vector<int> minimumCost = {0, 0};
    for (int i = 2; i < cost.size() + 1; ++i) {
      int takeOne = minimumCost[i - 1] + cost[i - 1];
      int takeTwo = minimumCost[i - 2] + cost[i - 2];
      minimumCost.push_back(std::min(takeOne, takeTwo));
    }
    return minimumCost.back();
  }
};
